File: GFG/LinkedList/delenddl.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

void delend(DLList<int> &dl){
    if(dl.head==NULL){
        return;
    }
    DNode<int>* curr=dl.head;
    while(curr->next->next!=NULL){
        curr=curr->next;
    }
    DNode<int>* temp=curr->next;
    curr->next=NULL;
    delete temp;
    return;
}
int main(){
    DLList<int> dll;
    dll.insert(5);
    dll.insert(10);
    dll.insert(15);
    dll.insert(20);
    delend(dll);
    dll.display();
    return 0;
}

File: GFG/LinkedList/delenddl.py
import ds

def delend(dl):
    if(dl.head is None):
        return
    curr=dl.head
    while(curr.next.next!=None):
        curr=curr.next
    curr.next=None
    
dll=ds.DLList()
dll.insert(5)
dll.insert(10)
dll.insert(15)
dll.insert(20)
delend(dll)
dll.display()